kod/object/active/holder/nomoveon/battler/player/user.kod

```diff
@@ -2399,29 +2413,90 @@ messages:
                OR oIllusion = $
                OR NOT IsClass(oIllusion,&Monster)
             {
-               oOtherGuild = Send(what,@GetGuild);
-               if poGuild <> $ AND oOtherGuild <> $
+               if IsClass(Send(self,@GetOwner),&BattleArena)
                {
-                  if poGuild = oOtherGuild
+                  % Spectators see the teams as different colors for viewing purposes.
+                  if Send(Send(self,@GetOwner),@GetUserTeam,#what=self) = 1
                   {
-                     iFlags = (iFlags | PLAYER_IS_GUILDMATE);
+                     if Send(Send(self,@GetOwner),@GetUserTeam,#what=what) = 2
+                     {
+                        if Send(Send(self,@GetOwner),@IsUserCarryingFlag,#what=what) = TRUE
+                        {
+                           iFlags = (iFlags | 0x0C000000);
```

Comment:
Shouldn't use hex values here, use a defined constant instead. Will try to get the bitflag expansion into this coming update so you can define more colors, and then this can go into 1.0.9.0.